evalml/pipelines/components/estimators/regressors/lightgbm_regressor.py

```diff
@@ -0,0 +1,96 @@
+import copy
+
+import pandas as pd
+from sklearn.preprocessing import OrdinalEncoder
+from skopt.space import Integer, Real
+
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components.estimators import Estimator
+from evalml.problem_types import ProblemTypes
+from evalml.utils import SEED_BOUNDS, get_random_seed, import_or_raise
+from evalml.utils.gen_utils import (
+    _convert_to_woodwork_structure,
+    _convert_woodwork_types_wrapper,
+    _rename_column_names_to_numeric
+)
+
+
+class LightGBMRegressor(Estimator):
+    """LightGBM Regressor"""
+    name = "LightGBM Regressor"
+    hyperparameter_ranges = {
+        "learning_rate": Real(0.000001, 1),
+        "boosting_type": ["gbdt", "dart", "goss", "rf"],
+        "n_estimators": Integer(10, 100),
+        "max_depth": Integer(0, 10),
+        "num_leaves": Integer(2, 100),
+        "min_child_samples": Integer(1, 100),
+        "bagging_fraction": Real(0.000001, 1),
+        "bagging_freq": Integer(0, 1)
+    }
+    model_family = ModelFamily.LIGHTGBM
+    supported_problem_types = [ProblemTypes.REGRESSION]
+
+    SEED_MIN = 0
+    SEED_MAX = SEED_BOUNDS.max_bound
+
+    def __init__(self, boosting_type="gbdt", learning_rate=0.1, n_estimators=100, max_depth=0, num_leaves=31, min_child_samples=20, n_jobs=-1, random_state=0, bagging_fraction=0.9, bagging_freq=0, **kwargs):
```

Comment:
@bchen1116 any particular reason for choosing these default values? It could be interesting to do some scanning on these parameters and see if they should be tweaked. `max_depth=0` jumps out to me as being too low--the default [in lightgbm is -1](https://lightgbm.readthedocs.io/en/latest/Parameters.html#max_depth) i.e. "no limit", but I don't think we should do that.